### DISCUSSION TITLE
How to recover a Longhorn RWX volume with corrupted filesystem

### DISCUSSION THREAD

**Discussion by kcao3** [longhorn/longhorn#4682] [Q&A]
Hello,

I currently have a Longhorn RWX volume with corrupted filesystem. From the logs of the share-manager-pvc-xxxx of this RWX volume, I can see errors like the following:
level=fatal msg="Error running start command: 'fsck' found errors on device /dev/longhorn/pvc-xxxx but could not correct them: fsck from util-linux 2.34\n/dev/longhorn/pvc-xxxx contains a file system with errors, check  forced.\n/dev/longhorn/pvc-xxxx: Resize inode not valid. \n\n/dev/longhorn/pvc-xxxx: UNEXPECTED  INCONSISTENCY; RUN fsck MANUALLY.\n\t(i.e., without -a or -p options)\n"

In the past I also ran into this same issue with other Longhorn RWO volumes but was able to resolve the issue by following this procedure: https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution.

However, I have a difficulty today while attempting to manually attach a Longhorn RWX volume to a node in the cluster and run fsck to fix its filesystem. I'm currently using Longhorn version 1.2.2. When I tried to attach a RWX volume from the Longhorn web UI, I only see this option:
"RWX volumes can only be manually attached in maintenance mode."  

If not quite sure what the maintenance mode really is. Before attempted to manually attach a RWX volume to a specific node, I did check and make sure nothing else was currently using this volume. When I tried to do so, it appears the web ui does allow me to attach the RWX volume to a specific host in the cluster. However, I notice there was NO such /dev/longhorn/pvc-xxxx device created on the host although the web UI indicates the RWX volume was currently being attached to such host.

In short, does anyone know how to manually attach a RWX volume to a specific node OR a procedure for recovering a Longhorn RWX volume  with corrupted filesystem?

Thank you in advance for your help and support!

**Comment by c3y1huang** [comment]
@joshimoo , @PhanLe1010 , any ideas?

**Comment by c3y1huang** [comment]
Posting for ref:  https://cloud-native.slack.com/archives/CNVPEL9U3/p1664931258378449?thread_ts=1664543756.230919&cid=CNVPEL9U3

From @joshimoo,
```
In this case the easiest approach is to change the access mode on the longhorn volume from RWX to RWO temporarily, that way you can manually attach the volume.
This should be doable via the longhorn UI, there is no need to change the PVC/PV or anything else just the volume access mode.
```

**Reply by kcao3** [reply]
First of all, thank you very much for your support. I am trying to temporarily change the Access Mode for a RWX volume to RWO from the longhorn UI. For some unknown reason, from the longhorn UI, the Update Access Mode option for this RWX volume is currently disabled/grayed out although the volume is currently in Detached state. Does anyone know why or how to enable this feature in the Longhorn UI?

**Reply by c3y1huang** [reply]
From @joshimoo , in the same slack thread:
```
Might be blocked on the UI side, you can also do it via the web interface against the API.
There is a call for updateAccessMode or alternatively via a kubectl edit/patch against the volume resource.

As long as the volume is detached that would be fine
```
